pkg/kvstore/etcd.go

```diff
@@ -520,21 +525,36 @@ func (e *etcdClient) renewSession(ctx context.Context) error {
 	// routines can get a lease ID of an already expired lease.
 	e.Lock()
 
-	timeoutCtx, cancel := context.WithTimeout(ctx, statusCheckTimeout)
-	defer cancel()
+	// Cancel any eventual old session context
+	if e.sessionCancel != nil {
+		e.sessionCancel()
+		e.sessionCancel = nil
+	}
+
+	// Create a context representing the lifetime of the session. It will
+	// timeout if the session creation does not succeed in time and then
+	// persists until:
+	//  - the parent context is cancelled due to the etcd client closing or
```

Comment:
Nit to be consistent with the rest of the comment: 

```suggestion
	//  - The parent context is cancelled due to the etcd client closing or
```